languages.adoc

```diff
@@ -18,88 +18,82 @@ SPDX-FileType: DOCUMENTATION
-* Each module *SHOULD* separate the implementation of the specification as defined in table <<folders>> below.
+[oft-sid="req~up-language-api~1"]
+--
+* *MUST* implement serializers, validators, and builders for the link:../basics/README.adoc[uProtocol data types] using the link:../up-core-api/README.adoc[up-core-api data model].
+--
 
+[oft-sid="req~up-language-transport-api~1"]
+--
+* *MUST* declare the language specific link:up-l1/README.adoc[uProtocol Transport Layer interface], such that the interface can be implemented by the transport libraries.
+--
 
-.Module Folders/Namespace
-[#folders,width="80%",cols="20%,80%",options="header"]
-|===
-| Folder | Description
+[oft-sid="req~up-language-comm-api~1"]
+--
+* *MUST* provide an implementation of the link:up-l2/README.adoc[uProtocol Communication Layer (uP-L2) `API].
+--
 
-| `validators`
-| static methods to validate the data model
+[oft-sid="req~up-language-naming~1"]
+--
+* *MUST* be named according to pattern `up-[LANGUAGE]`
+--
 
-| `serializer`
-| Any code that is responsible to Serialize and deserialize the data model
+[oft-sid="req~up-language-structure~1"]
+--
+* The top level namespace (package name) *MUST* contain `uprotocol` and *SHOULD* contain `org.eclipse.uprotocol` 
+--
 
-| `factory` (or `builder`)
-| Factory methods to build the data model per business logic (i.e. to build CloudEvents, UUIDs, etc..)
+[oft-sid="req~up-language-documentation~1"]
+--
+ * *MUST* contain a `README.adoc` that explains how to build and use the library. 
+--
 
-|===
+=== Test & CI
 
+[oft-sid="req~up-language-test~1"]
+--
+ * *MUST* be able to run the test suite such that merge requests (commits) cannot be merged till tests are passed
+--
 
-=== Modules
-
-Modules are packages (folders) that reside under the `src` directory. <<content>> table below lists the required modules.
-
-.SDK Modules
-[#content,width="100%",cols="20%,20%,60%",options="header"]
-|===
-| Folder | Specification |Description
-
-| `uri`
-| link:basics/uri.adoc[UUri]
-| How things are addressed in uProtocol
-
-| `uuid`
-| link:basics/uuid.adoc[UUID]
-| Unique ID & timestamp specification 
-
-| `cloudevents`
-| link:up-l1/cloudevents.adoc[CloudEvents]
-| Specification of how uProtocol data models can be expressed using CloudEvents data model
-
-| `transport`
-| link:up-l1/README.adoc[uTransport]
-a| uP-L1 Transport Interface & data model.
-
-* *MUST* declare the interface & data model where as link:../upclient.adoc[uProtocol Client Library] implement the said interface.
-
-| `rpc`
-| link:up-l2/rpcclient.adoc[uP-L2 RPC Client]
-a|APIs used by clients to invoke methods to simplify implementation of the RPC design pattern. 
-
-|===
-
+[oft-sid="req~up-language-ci-build~1"]
+--
+ * *MUST* have CI workflow that verifies pull requests compiles before they can be merged
+--
 
-== Test
+[oft-sid="req~up-language-ci-test~1"]
+--
+ * *SHOULD* cover 100% of all code and specifications defined in <<Source>>
```

Comment:
is this a requirement for the CI system or for the (unit) test suite itself?

Reply:
unit tests and then CI runs to ensure it is 100% covered.

Reply:
If I am not mistaken, the link target does not exist.

Reply:
Ok I fixed it, thanks for catching that